src/Webfactory/Slimdump/Database/Dumper.php

```diff
@@ -241,11 +192,11 @@ public function dumpData($table, Table $tableConfig, Connection $db, bool $noPro
 
         $wrappedConnection->setAttribute(PDO::MYSQL_ATTR_USE_BUFFERED_QUERY, true);
 
-        if ($bufferSize) {
+        if ($this->sqlDumper->getCurrentBufferSize()) {
```

Comment:
I think you need to push responsibilities a bit more to the `SqlDumper` here:

The `Dumper` class would just fetch the data and process it (null out values, mask, ...). 

Then, pass entire rows of data to the `SqlDumper`.

The `SqlDumper` is responsible for tracking the buffer size and to start new INSERT statements whenever necessary.